app/Http/Controllers/v1/EnrollmentController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Controllers\v1;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\StoreEnrollmentRequest;
+use App\Http\Requests\UpdateEnrollmentRequest;
+use App\Http\Resources\EnrollmentResource;
+use App\Models\Enrollment;
+use Illuminate\Http\Response;
+
+class EnrollmentController extends Controller
+{
+    public function index()
+    {
+        $perPage = request()->query('per_page', 10);
+        $sortBy = request()->query('sort', 'rank');
+        $sortOrder = request()->query('order', 'asc');
+        $search = request()->query('search', '');
+
+        $query = Enrollment::query();
+
+        // Apply search
+        if ($search) {
+            $query->where(function ($query) use ($search) {
+                $query->where('name', 'like', '%'.$search.'%')
+                    ->orWhere('links', 'like', '%'.$search.'%');
+            });
+        }
+
+        $query->orderBy($sortBy, $sortOrder);
+
+        $categories = $query->paginate($perPage);
+
+        return EnrollmentResource::collection($categories);
+    }
+
+    public function show($id)
+    {
+        $Enrollment = Enrollment::find($id);
+
+        if (! $Enrollment) {
+            return response()->json(['error' => 'Enrollment not found'], Response::HTTP_NOT_FOUND);
+        }
+
+        return new EnrollmentResource($Enrollment);
+    }
+
+    public function store(StoreEnrollmentRequest $request)
+    {
+        $validated = $request->validated();
+
+        $Enrollment = Enrollment::create($validated->all());
+
+        return EnrollmentResource::collection($Enrollment);
```

Comment:
Why are we returning a collection here? is this expecting creating/many enrollments at once?

Reply:
Thats how I did it in category, so i did the same thing for enrollments. I do not know if we will return many at once or not.

Reply:
as stated in slack, it's likely that the return json won't be used, and the frontend/consumer will simply be looking for a 201/created. i think worst case all it's doing is makes the front end have to index an additional field on the response. I'm fine with leaving it. this can be merged :+1: